clients/drcachesim/tracer/physaddr.cpp

```diff
@@ -60,14 +60,15 @@ static const addr_t PAGE_INVALID = (addr_t)-1;
 physaddr_t::physaddr_t()
 #ifdef LINUX
     : page_size_(dr_page_size())
-    , last_vpage_(PAGE_INVALID)
-    , last_ppage_(PAGE_INVALID)
+    , cache_idx_(0)
     , fd_(-1)
+    , v2p_(nullptr)
     , count_(0)
 #endif
 {
 #ifdef LINUX
-    v2p_.table = nullptr;
+    memset(last_vpage_, static_cast<char>(PAGE_INVALID), sizeof(last_vpage_));
```

Comment:
Does this work as expected? It will set the first `NUM_CACHE` bytes of `last_vpage_` to -1 right?

Reply:
sizeof applied to an array is the total size of all the elements.  So it will set `NUM_CACHE*sizeof(last_vpage_[0])` to -1.

Reply:
argh. I searched for size of and just read array::size. Sorry for the noise.

Reply:
If each byte has all the bits set (==-1), then the word will also have all the bits set and match PAGE_INVALID which is -1 cast to `addr_t==uintptr_t`.